include/highfive/bits/H5Easy_Eigen.hpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_EIGEN_HPP
+#define H5EASY_BITS_EIGEN_HPP
+
+#include "../H5Easy.hpp"
+#include "H5Easy_misc.hpp"
+#include "H5Easy_scalar.hpp"  // to get the basic "load_impl"
+
+#ifdef HIGHFIVE_EIGEN
+
+namespace HighFive {
+
+namespace detail {
+
+namespace eigen {
+
+// return the shape of the "Eigen::Matrix" as size 1 or 2 "std::vector<size_t>"
+template <class C, int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+std::vector<size_t> shape(const Eigen::Matrix<C,Rows,Cols,Options,MaxRows,MaxCols>& data)
+{
+    if (Rows == 1) {
+        return {static_cast<size_t>(data.cols())};
+    } else if (Cols == 1) {
+        return {static_cast<size_t>(data.rows())};
+    } else {
+        return {static_cast<size_t>(data.rows()), static_cast<size_t>(data.cols())};
+    }
+}
+
+// write to open DataSet of the correct size
+// if the "Eigen::Matrix" is column-major reordering has to be done first; HDF5 is always row-major
+template <class C, int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+void write(HighFive::DataSet& dataset, const Eigen::Matrix<C,Rows,Cols,Options,MaxRows,MaxCols>& data)
```

Comment:
This line is way too long. It is typically a bad idea to do "custom" formatting. Of course clang-format sometimes might generate ugly things that we can fix, but here I wonder what you had against one argument per line. That's standard in this project.